Weather

Ruslan1908 opened the pull request:
Добавлены файлы Header и Weather, изменены конфигурации и адаптированы другие файлы

Ruslan1908 committed:
Update App.jsx

Добавил Header

Ruslan1908 committed:
Update GymMap.jsx

Убрал import React

Ruslan1908 committed:
Update UserSetupForm.jsx

- import react

Ruslan1908 committed:
Update WorkoutScheduler.jsx

- import React

Ruslan1908 committed:
Update .gitignore

Пропуск проверки перед запуском для строки "skip_preflight_check"

Ruslan1908 committed:
Update .eslintrc.json

Обновленная конфигкрация

Ruslan1908 committed:
Add files via upload

Добавил файлы для хэдера и погоды

Ruslan1908 committed:
Update package.json

Ruslan1908 committed:
Update package-lock.json

AntonZlobichLambda commented on .gitignore:
```diff
@@ -21,3 +21,6 @@
 yarn-error.log*
+
+# Skip preflight check
+SKIP_PREFLIGHT_CHECK=true
```
Зачем этот параметр нужен?
И мне интересно, почему он в гитигноре?

Ruslan1908 replied:
Он отключает предварительные проверки. У меня была проблема с версией линта и чтобы приложение принудительно запустилось.
В gitgnore чтобы не создавать файл .env и на тройка оставалась локальной

AntonZlobichLambda replied:
Может решим проблему с версией линта?)

AntonZlobichLambda commented on src/App.jsx:
```diff
@@ -1,28 +1,71 @@
-    </Container>
+    <Router>
+      <div>
+        <Header>
```
А что если для какой-то из страниц нам понадобится другой header?
Или нам понадобится страница без headerа, например

AntonZlobichLambda replied:
Компонент типа layout с хедером (как пропс, или статично внутри layout)

AntonZlobichLambda commented on src/Header.jsx:
```diff
@@ -0,0 +1,29 @@
+  }
+`;
+
+export const Header = () => {
```
А зачем у тебя ещё второй такой же в `App.jsx`?

AntonZlobichLambda commented on src/Header.jsx:
```diff
@@ -0,0 +1,29 @@
+export const Header = () => {
+  return (
+    <HeaderContainer>
+      <NavLink to="/">Home</NavLink>
```
Что можно сделать, чтобы не дублировать постоянно по всему приложению ручками все пути?
(я имею в виду to/path и тд)

AntonZlobichLambda commented on src/Weather.jsx:
```diff
@@ -0,0 +1,58 @@
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  const lat = 33.44; 
```
Снова константы внутри компонента

AntonZlobichLambda commented on src/Weather.jsx:
```diff
@@ -0,0 +1,58 @@
+
+  const lat = 33.44; 
+  const lon = -94.04; 
+  const apiKey = '5fdd93652c06870916b3a6ccf6b88aca'; 
```
Такие штуки как api токены обычно хранят в `.env` файле и потом берут эту константу оттуда

AntonZlobichLambda commented on src/Weather.jsx:
```diff
@@ -0,0 +1,58 @@
+  useEffect(() => {
+    const fetchWeather = async () => {
+      try {
+        const response = await axios.get(
```
Мы можем вынести запрос в отдельную функцию в отдельный файл. Чтобы функция принимала только координаты или имя города и отдавала весь респонс. Это делается, чтобы вот такие здоровые urlы жили где-то отдельно

AntonZlobichLambda commented on src/Weather.jsx:
Комменты кажутся лишними. Или ты считаешь, что они нужны?

AntonZlobichLambda commented:
По файловой структуре уже ощущается необходимость в делении на разные папочки, а не просто всё в src

AntonZlobichLambda commented on src/App.jsx:
```diff
@@ -1,28 +1,71 @@
+
+
+const Header = styled.header`
+  position: fixed;
```
Почитай плез про правильное форматирование css, а именно про правильный порядок стилей

AntonZlobichLambda commented on src/App.jsx:
```diff
@@ -1,28 +1,71 @@
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 1000;
```
А у dialog'а будет `z-index: 10000`, а у тултипа `z-index: 1000000` :)
Ты очень часто будешь видеть такие значения у `z-index`, но это не есть идеально. В идеале значения `z-index` должны быть типа 1 или 2, без 999999.
Есть другие способы располагать элемент поверх другого (например оба на absolute и один ниже другого в вёрстке, тогда он будет поверх). Это не значит, что тут обязательно нужно применить другой, просто не стоит злоупотреблять `z-index: 99999`

AntonZlobichLambda replied:
Подумай, можно ли в этом случае его убрать

Ruslan1908 committed:
Delete src/App.css

Ruslan1908 committed:
Update App.jsx

исправил комменты

Ruslan1908 committed:
Update .gitignore

Ruslan1908 committed:
Update .eslintrc.json

обновил

Ruslan1908 committed:
Update Header.jsx

Ruslan1908 committed:
Update Weather.jsx

Ruslan1908 committed:
Add files via upload

Ruslan1908 committed:
Update Header.jsx

Ruslan1908 committed:
Add files via upload

Ruslan1908 committed:
Update Weather.jsx

AntonZlobichLambda commented on .gitignore:
```diff
@@ -1,23 +1,20 @@
 .env.test.local
 .env.production.local
 
+.env
```
Что-то пошло не так) .env, таки, виден в изменениях

AntonZlobichLambda commented on src/App.jsx:
```diff
@@ -1,28 +1,148 @@
+            <Route path="/weather" element={<Weather />} />
+          </Routes>
+        </MainContent>
+        {isDialogOpen && (
```
Не является частью текущей задачи, убираем

Ruslan1908 committed:
Update App.jsx

убран header

Ruslan1908 committed:
Update App.jsx

используется headerComponent

Ruslan1908 committed:
Update Header.jsx

Ruslan1908 committed:
Update Weather.jsx

вынесены константы
теперь вместо координат город

Ruslan1908 committed:
Update utils_weather.js

теперь город

Ruslan1908 committed:
Add files via upload

добавлены новые фалы

Ruslan1908 committed:
Delete .env

Ruslan1908 committed:
Create Header.jsx

Ruslan1908 committed:
Delete src/components/Header.jsx

Ruslan1908 committed:
Update and rename src/Header.jsx to src/components/Header.jsx

Ruslan1908 committed:
Update and rename HeaderComponent.jsx to src/components/HeaderComponent.jsx

Ruslan1908 committed:
Update and rename src/Weather.jsx to src/components/Weather.jsx

Ruslan1908 committed:
Update and rename src/WorkoutScheduler.jsx to src/components/WorkoutScheduler.jsx

Ruslan1908 committed:
Update and rename src/UserSetupForm.jsx to src/components/UserSetupForm.jsx

Ruslan1908 committed:
Update and rename src/GymMap.jsx to src/pages/GymMap.jsx

Ruslan1908 committed:
Rename src/index.css to src/styles/index.css

Ruslan1908 committed:
Update and rename src/firebase.js to src/services/firebase.js

Ruslan1908 committed:
Update and rename src/firebaseService.js to src/services/firebaseService.js

Ruslan1908 committed:
Update and rename src/utils_weather.js to src/utils/utils_weather.js

Ruslan1908 committed:
Update Header.jsx

Ruslan1908 committed:
Update UserSetupForm.jsx

Ruslan1908 committed:
Update Weather.jsx

Ruslan1908 committed:
Update App.jsx

Ruslan1908 committed:
Update index.js

AntonZlobichLambda commented on src/components/Header.jsx:
Контент идентичен App.jsx.
Плюс зачем этот компонент, если есть HeaderComponent?

AntonZlobichLambda commented on src/utils/utils_weather.js:
Папка для запросов обычно называется `api`. Так что у этого файла путь может быть типа `src/api/weather`

AntonZlobichLambda commented on routes.js:
```diff
@@ -0,0 +1,4 @@
+export const ROUTES = {
```
Эта константа пока не используется, нужно её использовать

AntonZlobichLambda commented on src/App.jsx:
```diff
@@ -1,28 +1,51 @@
+  padding-top: 60px;
+`;
+
+const Layout = ({ children }) => (
```
Layout всё ещё распространяется на все страницы.

Что я конкретно имел в виду по поводу Layout:

1) Вот этот компонент лежит в отдельном файле `Layout.jsx`

2) App.jsx выглядит как просто 
<img width="249" alt="image" src="https://github.com/user-attachments/assets/e6898126-21e3-4d39-a335-471813d5c457" />
И всё.

3) Каждая страница внутри обёрнута в этот Layout

<img width="377" alt="image" src="https://github.com/user-attachments/assets/f7ba8817-d8e1-4df9-87d7-55732e007302" />

Ruslan1908 committed:
Update App.jsx

1. Добавлен новый компонент WeatherWithLayout: Этот компонент оборачивает существующий Weather в Layout.
2ю Обновлён маршрут для страницы /weather.

Ruslan1908 committed:
Delete src/components/Header.jsx

не нужен

Ruslan1908 committed:
Update App.jsx

1.Обновлены пути, чтобы соответствовать константам в ROUTES.
2.Удалён MainContent. Заменён стиль содержимого на inline-стили для div-обертки

Ruslan1908 committed:
Update HeaderComponent.jsx

Ruslan1908 committed:
Add files via upload

Ruslan1908 committed:
Update and rename src/utils_weather.js to src/api/utils_weather.js

Ruslan1908 committed:
Delete src/utils/utils_weather.js

перемещен в api

Ruslan1908 committed:
Add files via upload

Ruslan1908 committed:
Update App.jsx

1. Компонент Layout в отдельном файле Layout.jsx.
2. В App.jsx маршруты определены через Router и Routes, а страницы вручную обёрнуты в Layout.
3ю Константа ROUTES используется для определения путей

Ruslan1908 committed:
Update HeaderComponent.jsx

Mаршруты ('/' и '/weather') были указаны явно. Теперь используются  значения из ROUTES

AntonZlobichLambda commented on routes.js:
```diff
@@ -0,0 +1,4 @@
+export const ROUTES = {
+    HOME: '/',
+    Weather: '/Weather',
```
Раз первый `HOME`, второй должен быть `WEATHER` (а не `Weather`)

AntonZlobichLambda commented on src/components/Weather.jsx:
```diff
@@ -0,0 +1,56 @@
+
+const DEFAULT_CITY = "Wroclaw"; // Город по умолчанию
+
+export const Weather = () => {
```
Должен лежать в pages, раз это страница

AntonZlobichLambda commented on src/components/Weather.jsx:
```diff
@@ -0,0 +1,56 @@
+  if (error) return <WeatherContainer>{error}</WeatherContainer>;
+
+  return (
+    <WeatherContainer>
```
Должен быть обёрнут ещё и в Layout

AntonZlobichLambda commented on src/App.jsx:
```diff
@@ -1,28 +1,42 @@
   return (
-    <Container maxWidth="sm">
-      <CssBaseline />
-      {!userData ? (
-        <UserSetupForm onSetupComplete={handleSetupComplete} />
-      ) : (
-        <>
-          <WorkoutScheduler sessionsPerWeek={userData.trainingPreferences.sessionsPerWeek} />
-          {userData.trainingPreferences.location === 'gym' && <GymMap />}
-        </>
-      )}
```
Вот это вот всё дело можно просто объединить внутри `<Route ... element={<Layout>{вот тут выделенный контент}</Layout>}`.
Или создать новую страницу в pages, типа `Home` и чтобы она рендерила (возвращала) `<Layout>{вот тут выделенный контент}</Layout>`. Но тогда надо придумать, как при переключении страниц не терять инфу с `UserSetupForm`

AntonZlobichLambda commented on src/api/utils_weather.js:
Файл назови просто `weather`.
Если есть папки типа `api` или `utils` или `constants`, то файлы в них называют просто `weather` или там `news`. Имя папки уже даёт понять, "о чём" будут файлы

Ruslan1908 committed:
Update routes.js

Ruslan1908 committed:
Update and rename Weather.jsx to Weather.jsx

обернул в layout
переместил в pages
